### DISCUSSION TITLE
flask-soketio The maximum number of connections can only be 512?

### DISCUSSION THREAD

**Discussion by liuqijie6** [miguelgrinberg/Flask-SocketIO#1920] [Q&A]
hardware： mac M1 。Could it be my computer's problem?

I'm testing the approximate number of connections.
Below is the code I tested.
Is there any way to increase the number of connections?

socketIo config
![image](https://user-images.githubusercontent.com/52588515/205829040-279481f2-8310-4c6c-9b8d-6cf2202d5a77.png)

```
@socket_io.on("connect")
def connect(kwargs):
    print(kwargs)

@socket_io.on('disconnect')
def disconnect_():
    sid = request.sid
    print(f'{sid} Client disconnected')

```
test code
<img width="870" alt="image" src="https://user-images.githubusercontent.com/52588515/205827387-1e7f1734-50ec-4fda-acf2-ba9e54a193d7.png">

pycharm console output
![image](https://user-images.githubusercontent.com/52588515/205827735-f767b129-0d5f-4fba-844f-3d28bed60bbe.png)
![image](https://user-images.githubusercontent.com/52588515/205828516-add782e6-1fb6-47c3-884a-c3bfb49d7516.png)

**Comment by miguelgrinberg** [comment]
This package has no explicit maximum. The limit is in how many file handles your process can create, and it is imposed by your operating system.

**Reply by liuqijie6** [reply]
I try to modify the mac maximum number of open files.
```
(base) fanren@fanrendeMBP ~ % ulimit -n
256
(base) fanren@fanrendeMBP ~ % ulimit -n 1024
(base) fanren@fanrendeMBP ~ % ulimit -n     
1024
```
But the result is still 512.

**Reply by miguelgrinberg** [reply]
The `ulimit` command is for UNIX based systems, I'm not sure it is the correct way to do this on BSD/macOS, never needed to reconfigure this myself.

**Reply by bluthen** [reply]
In my experience, you will probably run into other issues anyway after increasing it, so better to look at other ways to increase capacity if you are hitting the max file handles limit. The reason it may be hitting max file handles is that responses are taking too long, so increasing it wouldn't help that.

**Reply by miguelgrinberg** [reply]
They're hitting the max because they're trying to create 3000 connections in a for-loop, this isn't a realistic test or application.

**Reply by liuqijie6** [reply]
Simulation user connection。 Is there any problem with cycle creation?

**Reply by liuqijie6** [reply]
Because these connections are not closed。 So I think this can test the maximum number of connections

**Reply by miguelgrinberg** [reply]
It's not a great way to test. Unrelated from the issue of setting the max number of file handlers, this test is bad because it causes an unrealistic load on the server, which needs to perform 3000 connection handshakes in parallel or near parallel. You are unlikely to get as many connections simultaneously in a real world situation. You are also hosting your clients and your server on the same machine, so it's really 6000 connections. And you are creating 3000 threads in your Python client, which is a bit much.

**Reply by liuqijie6** [reply]
A user needs to create a connection.
Should I separate client and server? Two computers for testing?

**Reply by liuqijie6** [reply]
Or is there no problem on the server side, because the client cannot create too many threads and cause the connection to fail?
So it's because of the problem that the client cannot be created in the test code?

**Reply by miguelgrinberg** [reply]
All these decisions depend on what you need or want to simulate with this. There is actually no problems, you asked why this was a bad idea and I just gave you some reasons. You said you wanted to test how many connections can be made, and I answered that, you can have unlimited connections. The limit is not in this package, it is going to be in the resources that your application use.
